apps/dashboard/app/models/project.rb

```diff
@@ -199,6 +199,27 @@ def jobs
     end.flatten
   end
 
+  def readme_path
+    md_readme = "#{directory}/README.md"
+    return md_readme if File.exist?(md_readme) && File.readable?(md_readme)
+
+    txt_readme = "#{directory}/README.txt"
+    return txt_readme if File.exist?(txt_readme) && File.readable?(txt_readme)
+  end
+
+  def readme
+    file_content = File.read(readme_path)
+
+    if File.extname(readme_path) == '.md'
+      markdown = Redcarpet::Markdown.new(Redcarpet::Render::HTML)
+      return markdown.render(file_content).html_safe
```

Comment:
We have to be careful here with regards to `html_safe`.  I think we can use `sanitize` if we pull in this module: https://api.rubyonrails.org/classes/ActionView/Helpers/SanitizeHelper.html

Though that would suggest that it should be handled in a view or helper. In any case, here's a test markdown you can use. The fact that we can get `window.alert` to trigger shouldn't happen. These files need to be safe to distribute.

```markdown
# my cool project

<script>window.alert("this shouldn't be allowed");</script>
```